src/formatting.rs

```diff
@@ -58,6 +59,32 @@ impl<'b, T: Write + 'b> Session<'b, T> {
     }
 }
 
+/// Determine if a module should be skipped. True if the module should be skipped, false otherwise.
+fn should_skip_module<T: FormatHandler>(
+    config: &Config,
+    context: &FormatContext<'_, T>,
+    input_is_stdin: bool,
+    main_file: &FileName,
+    path: &FileName,
+    module: &Module<'_>,
+) -> bool {
+    if contains_skip(module.attrs()) {
+        return true;
+    }
+
+    let source_file = context.parse_session.span_to_file_contents(module.span);
+    let src = source_file.src.as_ref().expect("SourceFile without src");
+
+    let should_ignore = (!input_is_stdin && context.ignore_file(&path))
+        || (!config.format_generated_files() && is_generated_file(src));
+
+    if (config.skip_children() && path != main_file) || should_ignore {
+        return true;
+    }
+
+    false
+}
```

Comment:
Thanks for pulling this out, think it makes things much more clear. I realize what you've done here is identical to what was there before, though as I'm looking at this I feel like we could maybe clean it up a bit more?

For example, we could use the same type of early return pattern utilized with `contains_skip` in the function opening for a file based input that's ignored, and save the `span_to_file_contents` and `src.as_ref()` calls when those conditions are true

Reply:
I reorganized the ``should_skip_module`` function so now we only call ``span_to_file_contents`` and ``src.as_ref()`` when they are needed.